overlord/ifacestate/handlers.go

```diff
@@ -401,6 +402,10 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 	if err := task.Get("by-gadget", &byGadget); err != nil && err != state.ErrNoState {
 		return err
 	}
+	var delaySetupProfiles bool
+	if err := task.Get("delay-setup-profiles", &delaySetupProfiles); err != nil && err != state.ErrNoState {
```

Comment:
I would call the flag delayed-setup-profiles, the handler itself can only skip it, the delaying is something orchestrated more globally